components/global/core/io.cellery.observability.k8s.client/src/main/java/io/cellery/observability/k8s/client/ComponentsEventSource.java

```diff
@@ -0,0 +1,259 @@
+                }
+            }
+
+            @Override
+            public void onClose(KubernetesClientException cause) {
+                if (cause == null) {
+                    if (logger.isDebugEnabled()) {
+                        logger.debug("Kubernetes cell watcher closed successfully");
+                    }
+                } else {
+                    logger.error("Kubernetes cell watcher closed with error", cause);
+                }
+            }
+        });
+        if (logger.isDebugEnabled()) {
+            logger.debug("Created cell watcher");
+        }
+    }
+
+    @Override
+    public void disconnect() {
+        if (k8sClient != null) {
+            k8sClient.close();
+            if (logger.isDebugEnabled()) {
+                logger.debug("Closed API server client");
+            }
+        }
+    }
+
+    @Override
+    public void destroy() {
+        // Do nothing
+    }
+
+    @Override
+    public void pause() {
+        // Do nothing
+    }
+
+    @Override
+    public void resume() {
+        // Do nothing
+    }
+
+    @Override
+    public Map<String, Object> currentState() {
+        return null;
+    }
+
+    @Override
+    public void restoreState(Map<String, Object> map) {
+        // Do nothing
+    }
+
+    /**
+     * This method checks if the ingress type is of HTTP and adds it to the relevant cell data
+     */
+    public void addHTTPTypes(List<HTTP> httpObjectsList, SourceEventListener sourceEventListener,
+                             Map<String, Object> attributes, IngressTypes ingressTypes) {
+        if (httpObjectsList != null) {
+            for (HTTP httpObj : httpObjectsList) {
+                if (httpObj.getBackend() != null && !httpObj.getBackend().isEmpty()) {
+                    ingressTypes.addType(HTTP);
+                    attributes.put(INGRESS_TYPES, StringUtils.join(ingressTypes.getTypes(), ','));
+                    attributes.put(Constants.ATTRIBUTE_COMPONENT, httpObj.getBackend());
+                    sourceEventListener.onEvent(attributes, new String[0]);
+                }
+            }
+        }
+    }
+
+    /**
+     * This method checks if the ingress type is of TCP and adds it to the relevant cell data
+     */
+    public void addTCPTypes(List<TCP> tcpObjectsList, SourceEventListener sourceEventListener,
+                            Map<String, Object> attributes, IngressTypes ingressTypes) {
+        if (tcpObjectsList != null) {
+            for (TCP tcpObject : tcpObjectsList) {
+                if (tcpObject.getBackendHost() != null && !tcpObject.getBackendHost().isEmpty()) {
+                    ingressTypes.addType(TCP);
+                    attributes.put(INGRESS_TYPES, StringUtils.join(ingressTypes.getTypes(), ','));
+                    attributes.put(Constants.ATTRIBUTE_COMPONENT, tcpObject.getBackendHost());
+                    sourceEventListener.onEvent(attributes, new String[0]);
```

Comment:
Same problem commented above also applies here.

Reply:
Noted. I will update the logic.